### DISCUSSION TITLE
fixing invalid bip39 mnemonic

### DISCUSSION THREAD

**Discussion by gorazdko** [BlockchainCommons/Gordian-Developer-Community#27] [Ideas]
## Abstract

Specter-DIY recently implemented it in https://github.com/cryptoadvance/specter-diy/pull/86

As @ChristopherA sums it up
> The goal here is to allow for seed creation by using dice to words manual techniques that include adding a last word, then based on the randomness of all 12 (or 24 words) such that includes the randomness of the last word, there should be exactly 1 checksum that is the same no matter the implementation. (if we did this with 11 or 23 words, we loose a few bits of randomness, and as there are multiple valid checksums possible making is less deterministic).

> In particular I'm uncomfortable with the word "fix". There nothing wrong with the last word if you use this technique, you are just choosing to let the app complete the checksum. Maybe "complete"? "mend"? I'm not sure. It is possible that there maybe is no other verb or phrase than "fix"

> Ideally all the implementations should use the same verb or phrase in all the implementations, i.e. spectre (@stepansnigirev), seedtool (@wolfmcnally), gordian (@Fonta1n3), proofwallet (@hodlwave), and maybe someday even the BIP39 javascript version (@iancoleman) so that someone can confirm no "up my sleeves" tricks with this last word on all the implementations.

So this issue is here to come up with a name/verb for this technique. This would allow for people to verify the outcome of the procedure against different manuals written by different people.

Since this technique differs from [Rudefox ShowMyWork](https://www.rudefox.io/blog/2020-07-16-show-my-work.html) and [Seedpicker](https://github.com/merland/seedpicker) in that the checksum word is included in the randomized process it introduces additional risks which can be documented here: e.g. user backups the invalid mnemonic sentence instead of the adapted one

(discussion moved from https://github.com/BlockchainCommons/bc-lethekit/pull/58)

**Comment by ChristopherA** [comment]
There is some other discussion in an PR to implement this feature in LetheKit https://github.com/BlockchainCommons/bc-lethekit/pull/58

**Comment by Fonta1n3** [comment]
Validate mnemonic? (As it is invalid without the checksum)

**Reply by Fonta1n3** [reply]
or Finalize?

**Comment by lucasmoten** [comment]
How about "Close" (Close invalid bip39 mnemonic) or "Closure", as in the "Mnemonic Seed Closure".  This can also be used when referring to the final checksum driven "Closure Word"

**Reply by lucasmoten** [reply]
Additional terms to consider at https://www.wordhippo.com/what-is/another-word-for/final_word.html

**Comment by RandyMcMillan** [comment]
postamble

**Reply by RandyMcMillan** [reply]
a deterministic postamble

**Reply by RandyMcMillan** [reply]
d'amble
di-amble
